// option java_multiple_files = true;
// option java_package = "io.grpc.examples.helloworld";
// option java_outer_classname = "HelloWorldProto";
// option objc_class_prefix = "HLW";

syntax = "proto3";

package clipper.grpc;

option cc_enable_arenas = true;

service Predict {
  rpc Predict(PredictRequest) returns (PredictResponse) {
  }
}

message ByteData {
    bytes data = 1;
}

message IntData {
    repeated int32 data = 1;
}

message FloatData {
    repeated float data = 1;
}

message DoubleData {
    repeated double data = 1;
}

message StringData {
    string data = 1;
}

message PredictRequest {
  string application = 1;
  int32 data_type = 2;
  ByteData byte_data = 3;
  IntData int_data = 4;
  FloatData float_data = 5;
  DoubleData double_data = 6;
  StringData string_data = 7;
}

message PredictResponse {
  bool has_error = 1;
  string error = 2;
  bool is_default = 3;
  string default_explanation = 4;
  int32 data_type = 5;
  ByteData byte_data = 6;
  IntData int_data = 7;
  FloatData float_data = 8;
  StringData string_data = 9;
}
